### DISCUSSION TITLE
How to disable the body shifter ?

### DISCUSSION THREAD

**Discussion by elaatifi** [unpoly/unpoly#610] [General]
Hello,

We have this app using unpoly and all pages are not all of the same height so when moving from a page to the other the whole layout shift (because of the scrollbars). The fix has been to use : 
```css
html {
    scrollbar-gutter: stable;
}
```
No more flickering between pages. page transitions are so smooth all third party CSS has been fixed except Unpoly keep injecting the right-padding I could not find a way to disable this behaviour. The issue is Bootstrap and other third party scripts are using a quite polite css class added to body or html element but Unpoly was quite rude by setting style attribute and no clear way to to disable it.

The only thing I could do is to patch it using:
```js
up.BodyShifter.prototype._changeStyle = function () {
}
```

But this workaround hurt my eyes more than the shifting layout so if someone has any idea how to get rid of this would be much appreciated.

BTW Unpoly is amazing !

**Comment by triskweline** [comment]
Unpoly 3.8 will let elements control their appearance while a scrolling overlay is open.

![up-scrollbar-away](https://github.com/unpoly/unpoly/assets/89953/42fbd435-61e2-460c-b83a-5b33a80de029)

**Reply by elaatifi** [reply]
Good news :) Thank you very much !

**Reply by triskweline** [reply]
This is now available in [Unpoly 3.8.0](https://github.com/unpoly/unpoly/discussions/644).
